### PR TITLE
systemd.special.xml: correct info about implicit depends

### PR DESCRIPTION
**Pull request by msoltyspl**
Small corrections/updates about implicit dependencies for basic.target,
sysinit.target and shutdown.target.

This patch makes systemd.special(7) consistent with systemd.service(5) and systemd's actual behaviour.

### CODE REVIEW

File: man/systemd.special.xml [hunk 1]
```diff
@@ -512,7 +512,8 @@
           shall add <varname>Conflicts=</varname> dependencies to this
           unit for their service unit, which is implicitly done when
           <varname>DefaultDependencies=yes</varname> is set (the
-          default).</para>
+          default) along with <varname>Before=</varname> dependency.
+          </para>
```
Comment (arvidjaar) [inline man/systemd.special.xml:516]: Should not it rather recommend the same, as systemd does implicitly - adding Before together with Conflicts? If there is only Conflicts, but no Before/After, then there is no ordering between two jobs and it may in principle happen that stopping of unit happens too late.

Comment (msoltyspl) [inline man/systemd.special.xml:516]: So basically reword it to follow systemd implicit behavior. Ok !

BTW, I found out that .target units also get implicit dependency Conflicts=shutdown.target, but without Before=. But it doesnt seem to be mentioned anywhere (at least not in systemd.target, nor in systemd.special). Should I add mention about this as well ?

**Comment by msoltyspl** [comment]
Anyway, will prepare v2.
